lib/trino-orc/src/test/java/io/trino/orc/TestWriteBloomFilter.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.orc;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.slice.Slice;
+import io.trino.orc.metadata.Footer;
+import io.trino.orc.metadata.OrcMetadataReader;
+import io.trino.orc.metadata.OrcType;
+import io.trino.orc.metadata.Stream;
+import io.trino.orc.metadata.StripeFooter;
+import io.trino.orc.metadata.StripeInformation;
+import io.trino.orc.stream.OrcChunkLoader;
+import io.trino.orc.stream.OrcInputStream;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.Type;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static io.trino.orc.OrcTester.READER_OPTIONS;
+import static io.trino.orc.metadata.CompressionKind.NONE;
+import static io.trino.orc.metadata.Stream.StreamKind.BLOOM_FILTER_UTF8;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.lang.Math.toIntExact;
+import static org.testng.Assert.assertEquals;
+
+public class TestWriteBloomFilter
```

Comment:
The triggering conditions for the bug seems to be dictionary becoming too large and the writer falling back to non-dictionary encoding. But I don't see any tweaks to orc writer config (like reducing hive.orc.writer.dictionary-max-memory) which would guarantee that the conditions of the bug are reproduced.
It seems to me that it would be simpler to just test this in TestSliceDictionaryColumnWriter for CHAR, VARCHAR and STRING types and just call `tryConvertToDirect` and `getBloomFilters` directly in the unit test.